src/util/schnorr.rs

```diff
@@ -128,12 +128,11 @@ impl TapTweak for UntweakedKeyPair {
     ///
     /// # Returns
     /// The tweaked key and its parity.
-    fn tap_tweak<C: Verification>(self, secp: &Secp256k1<C>, merkle_root: Option<TapBranchHash>) -> TweakedKeyPair {
+    fn tap_tweak<C: Verification>(mut self, secp: &Secp256k1<C>, merkle_root: Option<TapBranchHash>) -> TweakedKeyPair {
         let pubkey = XOnlyPublicKey::from_keypair(&self);
         let tweak_value = TapTweakHash::from_key_and_tweak(pubkey, merkle_root).into_inner();
-        let mut output_key = self.clone();
-        output_key.tweak_add_assign(&secp, &tweak_value).expect("Tap tweak failed");
-        TweakedKeyPair(output_key)
+        self.tweak_add_assign(&secp, &tweak_value).expect("Tap tweak failed");
```

Comment:
I don't mean to de-reail your PR but why is it ok to panic here please? The docs of this function say that this can error if the resulting key would be invalid, is that not possible here?

Reply:
It will fail with negligible probability - multiple orders if magnitude less then probability of meteor hitting the server directly. 

It can only fail if the tweak * G is the inverse of the original public key - which could not happen with hash computed out of key itself - or if the tweak is > than group order, which has that "several orders of magnitude of meteor hitting server" probability.

Reply:
Perhaps the panic message should've been "Tap tweak broken" since that's a more likely cause for the error. (We had similar discussion not long ago.)

I wouldn't mind if the method itself just panicked without returning `Result` but that's another discussion.